time_tracker_api/time_entries/time_entries_model.py

```diff
@@ -74,6 +74,35 @@ def create_sql_ignore_id_condition(id: str):
         else:
             return "AND c.id!=@ignore_id"
 
+    @staticmethod
+    def create_sql_date_range_filter(custom_args: dict) -> str:
+        if 'start_date' and 'end_date' in custom_args:
+            return """
+            ((c.start_date BETWEEN @start_date AND @end_date) OR
+             (c.end_date BETWEEN @start_date AND @end_date))
+            """
+        else:
+            return ''
+
+    def find_all(self, partition_key_value: str, conditions: dict, custom_args: dict):
```

Comment:
The find_all in time entries should be more personalized. This function can and should be aware of parameters related to the month and year you want to query (you can create 2 params or better one for this kind of filtering). That will be better than `custom_args`, which is ambiguous for this purpose.